src/core/space_interaction_API.jl

```diff
@@ -192,12 +227,22 @@ end
 #######################################################################################
 # %% Space agnostic adding
 #######################################################################################
+
+function add_agent_to_model!(agent, model::ABM{<:SpaceType,A,Dict{Int, A}}) where {A<:AbstractAgent}
+    model.agents[agent.id] = agent
+end
+
+function add_agent_to_model!(agent, model::ABM{<:SpaceType,A,Vector{A}}) where {A<:AbstractAgent}
+    agent.id == nagents(model) + 1 || error("Cannot add agent of ID $(agent.id) in a vector ABM of $(nagents(model)) agents. Expected ID == $(nagents(model)+1).")
+    push!(model.agents, agent)
+end
+
 """
     add_agent_pos!(agent::AbstractAgent, model::ABM) → agent
 Add the agent to the `model` at the agent's own position.
 """
 function add_agent_pos!(agent::AbstractAgent, model::ABM)
-    model[agent.id] = agent
+    add_agent_to_model!(agent, model)
     model.maxid[] < agent.id && (model.maxid[] = agent.id)
     add_agent_to_space!(agent, model)
```

Comment:
The `model.maxid` alteration **must** happen in the `add_agent_to_model!` call. Not outside it. This makes for safer code.

Reply:
Done.